packages/ui-radio-input/src/RadioInput/index.tsx

```diff
@@ -22,151 +22,199 @@
+            return
+          }
+
+          if (typeof checkedProp === 'undefined') {
+            setInternalChecked(!internalChecked)
+          }
+
+          if (typeof onChange === 'function') {
+            onChange(e)
+          }
+        },
+        [disabled, readOnly, checkedProp, internalChecked, onChange]
+      )
+
+    // Expose imperative API via ref
+    useImperativeHandle(
+      ref,
+      () => ({
+        focus: () => {
+          inputElementRef.current?.focus()
+        },
+        get focused() {
+          return isActiveElement(inputElementRef.current)
+        },
+        get checked() {
+          return checked
+        },
+        get id() {
+          return id
+        }
+      }),
+      [checked, id]
+    )
 
-    if (typeof this.props.checked === 'undefined') {
-      this.setState({ checked: !this.state.checked })
+    const handleMouseOver = () => {
+      setHovered(true)
     }
 
-    if (typeof this.props.onChange === 'function') {
-      this.props.onChange(e)
+    const handleMouseOut = () => {
+      setHovered(false)
     }
-  }
-
-  focus() {
-    this._input?.focus()
-  }
-
-  get id() {
-    return this.props.id || this._defaultId
-  }
-
-  get focused() {
-    return isActiveElement(this._input)
-  }
-
-  get checked() {
-    return typeof this.props.checked === 'undefined'
-      ? this.state.checked
-      : this.props.checked
-  }
-
-  render() {
-    const { disabled, readOnly, label, value, name, styles } = this.props
-
-    const props = omitProps(this.props, RadioInput.allowedProps)
 
+    const passedProps = passthroughProps(rest)
     return (
       <div
         css={styles?.radioInput}
         data-cid="RadioInput"
-        ref={(el) => {
-          this.ref = el
-        }}
+        ref={containerRef}
+        /* eslint-disable-next-line jsx-a11y/mouse-events-have-key-events */
+        onMouseOver={handleMouseOver}
+        /* eslint-disable-next-line jsx-a11y/mouse-events-have-key-events */
+        onMouseOut={handleMouseOut}
       >
-        <div css={styles?.container}>
-          <input
-            {...props}
-            id={this.id}
-            ref={this.handleInputRef}
-            value={value}
-            name={name}
-            checked={this.checked}
-            type="radio"
-            css={styles?.input}
-            disabled={disabled || readOnly}
-            onChange={this.handleChange}
-            onClick={this.handleClick}
-          />
-          <label css={styles?.control} htmlFor={this.id}>
-            <span css={styles?.facade} aria-hidden="true" />
-            <span css={styles?.label}>{label}</span>
-          </label>
-        </div>
+        <input
+          {...passedProps}
+          id={id}
+          ref={handleInputRef}
+          value={value}
+          name={name}
+          checked={checked}
+          type="radio"
+          css={styles?.input}
+          readOnly={readOnly}
+          disabled={disabled}
```

Comment:
I'm not sure, that this is a good change. Why did the old code just make `readonly` disabled?